src/state_machine.zig

```diff
@@ -1861,45 +1969,53 @@ pub fn StateMachineType(
             if (dr_account.debits_exceed_credits(amount)) return .exceeds_credits;
             if (cr_account.credits_exceed_debits(amount)) return .exceeds_debits;
 
-            // After this point, the transfer must succeed.
-            defer assert(self.commit_timestamp == t.timestamp);
+            // After this point, the transfer must succeed atomically.
+            var result: CreateTransferResult = undefined;
+            defer assert(result == .ok);
+            defer if (operation == .create_transfers) assert(self.commit_timestamp == t.timestamp);
+            result = completed_atomically: {
```

Comment:
What is the reason for this refactor?

Reply:
```diff
- defer assert(self.commit_timestamp == t.timestamp);
```

We don't update `self.commit_timestamp` for import_* operations. Since we are dealing with past timestamps, the commit_timestamp is kept, like in lookups and queries.

Reply:
:thinking: I don't understand how that is connected to the `completed_atomically` blocks.

Also, I think that due to https://github.com/tigerbeetle/tigerbeetle/pull/1989#discussion_r1683206981 you may as well update `commit_timestamp` anyway.